gateway-ha/src/main/java/io/trino/gateway/ha/router/IRoutingRulesManager.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.router;
+
+import io.trino.gateway.ha.persistence.dao.RoutingRule;
+
+import java.util.List;
+
+public interface IRoutingRulesManager
```

Comment:
I will rename this to `RoutingRulesManager` once reviews are complete. I have not renamed yet because I renamed the existing `RoutingRulesManager` to `FileBasedRoutingRulesManager`, and made changes to the file. Git cannot track a name change followed by creating a new file with the old name, so performing both in the same commit makes the diffs useless.